jetty-util/src/main/java/org/eclipse/jetty/util/URIUtil.java

```diff
@@ -120,7 +120,7 @@ private static StringBuilder encodePath(StringBuilder buf, String path, int offs
                         buf = new StringBuilder(path.length() * 2);
                         break loop;
                     default:
-                        if (c > 127)
+                        if (c < 20 || c > 127)
```

Comment:
Decimal `20` is not good I guess? If you means the space it's decimal `32`, or hex `0x20`.

Reply:
ooops.  Added a better test that checks the whole range of characters.  Also encode 127, which is DEL!